dynawo/sources/Modeler/Common/DYNModelMulti.cpp

```diff
@@ -484,23 +484,24 @@ bool
 ModelMulti::propagateZModif() {
   vector<int> indicesDiff;
   silentZChange_ = false;
-  for (int i = 0; i < sizeZ(); ++i) {
-    bool isSilent = silentZ_[i];
-    if (isSilent && silentZChange_) {
-      continue;
-    } else if (isSilent) {
-      silentZChange_ =  doubleNotEquals(zLocal_[i], zSave_[i]);
-    } else if (doubleNotEquals(zLocal_[i], zSave_[i])) {
-      indicesDiff.push_back(i);
+  for (int i = 0, iEnd = nonSilentZIndexes_.size(); i < iEnd; ++i) {
+    if (doubleNotEquals(zLocal_[nonSilentZIndexes_[i]], zSave_[nonSilentZIndexes_[i]])) {
+      indicesDiff.push_back(nonSilentZIndexes_[i]);
     }
   }
 
   bool changeDetected = !indicesDiff.empty();
   if (changeDetected) {
     connectorContainer_->propagateZDiff(indicesDiff, zLocal_);
     std::copy(zLocal_, zLocal_ + sizeZ(), zSave_.begin());
-  } else if (silentZChange_) {
-    std::copy(zLocal_, zLocal_ + sizeZ(), zSave_.begin());
+  } else {
+    for (int i = 0, iEnd = silentZIndexes_.size(); i < iEnd; ++i) {
+      if (doubleNotEquals(zLocal_[silentZIndexes_[i]], zSave_[silentZIndexes_[i]])) {
+        silentZChange_ = true;
+        std::copy(zLocal_, zLocal_ + sizeZ(), zSave_.begin());
```

Comment:
I don't know if there would be any gain but as we have the indexes we can copy only the needed ones (ie, the silentZIndexes).

Reply:
std::copy is very efficient and multi-threaded, so it would probably be longer to copy only a subpart

Reply:
I was expected this ;). Fine for me then!